File: aspnet-core/src/GYISMS.Core/MeetingRooms/MeetingRoom.cs
using Abp.Domain.Entities;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GYISMS.MeetingRooms
{
    /// <summary>
    /// 会议室
    /// </summary>
    [Table("MeetingRooms")]
    public class MeetingRoom : Entity<int>
    {
        /// <summary>
        /// 会议室名称
        /// </summary>
        [Required]
        [StringLength(50)]
        public virtual string Name { get; set; }

        /// <summary>
        /// 会议室图片
        /// </summary>
        [Required]
        [StringLength(200)]
        public virtual string Photo { get; set; }

        /// <summary>
        /// 坐席人数
        /// </summary>
        [Required]
        public virtual int Num { get; set; }

        /// <summary>
        /// 会议室类型（固定会议室、临时会议室）
        /// </summary>
        public virtual int? RoomType { get; set; }

        /// <summary>
        /// 地址
        /// </summary>
        [StringLength(500)]
        public virtual string Address { get; set; }

        /// <summary>
        /// 楼层信息（如：A栋1单元20层2003）
        /// </summary>
        public virtual string BuildDesc { get; set; }

        /// <summary>
        /// 是否需要审批
        /// </summary>
        public virtual bool? IsApprove { get; set; }

        /// <summary>
        /// 会议室管理员Id（需要审批 必须添加管理员）
        /// </summary>
        public virtual string ManagerId { get; set; }

        /// <summary>
        /// 会议室管理员名称
        /// </summary>
        [StringLength(50)]
        public virtual string ManagerName { get; set; }

        /// <summary>
        /// 第几排
        /// </summary>
        public virtual int? Row { get; set; }

        /// <summary>
        /// 第几号（从左到右边）
        /// </summary>
        public virtual int? Column { get; set; }

        /// <summary>
        /// 会议室布局（中心模式、矩阵模式）
        /// </summary>
        public virtual int? LayoutPattern { get; set; }

        /// <summary>
        /// 会议室平面图（平面图需提前发我们做处理）
        /// </summary>
        public virtual string PlanPath { get; set; }

        /// <summary>
        /// 备注
        /// </summary>
        [StringLength(500)]
        public virtual string Remark { get; set; }

        /// <summary>
        /// 会议室设备配置名称（逗号分隔）
        /// </summary>
        [StringLength(500)]
        public virtual string Devices { get; set; }

        /// <summary>
        /// 是否删除
        /// </summary>
        public virtual bool? IsDeleted { get; set; }

        /// <summary>
        /// CreationTime
        /// </summary>
        public virtual DateTime? CreationTime { get; set; }

        /// <summary>
        /// CreatorUserId
        /// </summary>
        public virtual long? CreatorUserId { get; set; }

        /// <summary>
        /// LastModificationTime
        /// </summary>
        public virtual DateTime? LastModificationTime { get; set; }

        /// <summary>
        /// LastModifierUserId
        /// </summary>
        public virtual long? LastModifierUserId { get; set; }

        /// <summary>
        /// DeletionTime
        /// </summary>
        public virtual DateTime? DeletionTime { get; set; }

        /// <summary>
        /// DeleterUserId
        /// </summary>
        public virtual long? DeleterUserId { get; set; }
    }
}

File: ng-admin/src/shared/entity/meeting-management/meeting-room.ts
export class MeetingRoom implements IMeetingRoom {
    id: number;
    name: string;
    photo: string;
    num: number;
    roomType: number;
    address: string;
    buildDesc: string;
    isApprove: boolean;
    managerId: string;
    managerName: string;
    row: number;
    column: number;
    layoutPattern: number;
    planPath: string;
    remark: string;
    devices: string;
    isDeleted: boolean;
    creationTime: Date;
    creatorUserId: number;
    lastModificationTime: Date;
    lastModifierUserId: number;
    deletionTime: Date;
    deleterUserId: number;
    showPhoto: string;
    layoutName: string;
    typeName: string;

    constructor(data?: IMeetingRoom) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    init(data?: any) {
        if (data) {
            this.id = data["id"];
            this.name = data["name"];
            this.photo = data["photo"];
            this.num = data["num"];
            this.roomType = data["roomType"];
            this.address = data["address"];
            this.buildDesc = data["buildDesc"];
            this.isApprove = data["isApprove"];
            this.managerId = data["managerId"];
            this.managerName = data["managerName"];
            this.row = data["row"];
            this.column = data["column"];
            this.layoutPattern = data["layoutPattern"];
            this.planPath = data["planPath"];
            this.remark = data["remark"];
            this.devices = data["devices"];
            this.isDeleted = data["isDeleted"];
            this.creationTime = data["creationTime"];
            this.creatorUserId = data["creatorUserId"];
            this.lastModificationTime = data["lastModificationTime"];
            this.lastModifierUserId = data["lastModifierUserId"];
            this.deletionTime = data["deletionTime"];
            this.deleterUserId = data["deleterUserId"];
            this.layoutName = data["layoutName"];
            this.typeName = data["typeName"];
        }
    }

    static fromJS(data: any): MeetingRoom {
        let result = new MeetingRoom();
        result.init(data);
        return result;
    }

    static fromJSArray(dataArray: any[]): MeetingRoom[] {
        let array = [];
        dataArray.forEach(result => {
            let item = new MeetingRoom();
            item.init(result);
            array.push(item);
        });

        return array;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["id"] = this.id;
        data["name"] = this.name;
        data["photo"] = this.photo;
        data["num"] = this.num;
        data["roomType"] = this.roomType;
        data["address"] = this.address;
        data["buildDesc"] = this.buildDesc;
        data["isApprove"] = this.isApprove;
        data["managerId"] = this.managerId;
        data["managerName"] = this.managerName;
        data["row"] = this.row;
        data["column"] = this.column;
        data["layoutPattern"] = this.layoutPattern;
        data["planPath"] = this.planPath;
        data["remark"] = this.remark;
        data["devices"] = this.devices;
        data["isDeleted"] = this.isDeleted;
        data["creationTime"] = this.creationTime;
        data["creatorUserId"] = this.creatorUserId;
        data["lastModificationTime"] = this.lastModificationTime;
        data["lastModifierUserId"] = this.lastModifierUserId;
        data["deletionTime"] = this.deletionTime;
        data["deleterUserId"] = this.deleterUserId;
        data["layoutName"] = this.layoutName;
        data["typeName"] = this.typeName;
        return data;
    }

    clone() {
        const json = this.toJSON();
        let result = new MeetingRoom();
        result.init(json);
        return result;
    }
}
export interface IMeetingRoom {
    id: number;
    name: string;
    photo: string;
    num: number;
    roomType: number;
    address: string;
    buildDesc: string;
    isApprove: boolean;
    managerId: string;
    managerName: string;
    row: number;
    column: number;
    layoutPattern: number;
    planPath: string;
    remark: string;
    devices: string;
    isDeleted: boolean;
    creationTime: Date;
    creatorUserId: number;
    lastModificationTime: Date;
    lastModifierUserId: number;
    deletionTime: Date;
    deleterUserId: number;
    showPhoto: string;
    layoutName: string;
    typeName: string;
}

export class CheckBoxList implements ICheckBoxList {
    label: string;
    value: string;
    checked: boolean;
    constructor(data?: ICheckBoxList) {
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    init(data?: any) {
        if (data) {
            this.label = data["label"];
            this.value = data["value"];
        }
    }

    static fromJS(data: any): CheckBoxList {
        let result = new CheckBoxList();
        result.init(data);
        return result;
    }

    static fromJSArray(dataArray: any[]): CheckBoxList[] {
        let array = [];
        dataArray.forEach(result => {
            let item = new CheckBoxList();
            item.init(result);
            array.push(item);
        });
        return array;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        data["label"] = this.label;
        data["value"] = this.value;

        return data;
    }

    clone() {
        const json = this.toJSON();
        let result = new CheckBoxList();
        result.init(json);
        return result;
    }
}
export interface ICheckBoxList {
    label: string;
    value: string;
    checked: boolean;
}